src/views/Message/CustomForm.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+import { jwt } from "../../api/state";
+import { useRecoilState } from "recoil";
+import { profile, application } from "../../api/state";
+import { useRecoilValue } from "recoil";
+import { useHistory } from "react-router-dom";
+import Loading from "../Message/Loading";
+import Form from "../../views/Form";
+
+const CustomForm = () => {
+  const [typeformID, setTypeformID] = useState("NxomQTLD");
+  const { isLoading, isAuthenticated, getAccessTokenSilently } = useAuth0();
+  const [token, setToken] = useRecoilState(jwt);
+  const history = useHistory();
+  const user_profile = useRecoilValue(profile);
+  const apps = useRecoilValue(application);
+
+  useEffect(() => {
+    const fn = async () => {
+      if (isAuthenticated && !token.isSet) {
+        setToken({ token: await getAccessTokenSilently(), isSet: true });
+      }
+    };
+    fn();
+  }, [isLoading, isAuthenticated, token, setToken, getAccessTokenSilently]);
+
+  useEffect(() => {
+    if (isLoading || user_profile.exists || user_profile.isLoading) {
+      return;
+    }
+    if (isAuthenticated) {
+      history.push("/newprofile");
+    }
+  }, [isLoading, isAuthenticated, user_profile, history]);
+
+  useEffect(() => {
+    console.log(apps);
+    if (apps.total as number > 0) {
```

Comment:
Why do we use app.total, a list of applications, as a Recoil value? What are the particular benefits to doing it in this manner, with regards to how it may re-render the state?

Reply:
Two reasons:
 - OpenAPI specifications require that any api endpoints that return an array of values should also return a separate value that contains the length of the array. Its considered "good" programming practice in API development because anything on the frontend that might require pagination uses that value.
 - The main reason I use it here as a sort of "validation" check is because if you look at the backend where the applications are fetched the `catch(err)` block returns 0 as the number of applications in the event that an error occurs. Thus, if in some situation there's bad data passed in or something else throws an error we can expect that the default value of `apps.total` to be 0. The only way its a number >0 would be if things worked out well and there are actual type forms to render.

Reply:
```
.forEach((app: application_interface) => {
           setTypeformID(app.typeform_id);
         });
```

this is the part that results in the component getting re-rendered. Initially the CustomForm component is designed to render the `<Loading />` component. However, if we find a match for the endpoint (meaning there exists a different typeform to render on this page) then we call `setTypeformID` and that will result in the component re-rendering because the render function makes use of this variable to determine whether to display the loading page or a typeform.